crates/datis-core/src/extract.rs

```diff
@@ -51,58 +51,120 @@ pub fn extract_atis_station_frequencies(situation: &str) -> HashMap<String, Stat
 }
 
 pub fn extract_atis_station_config(config: &str) -> Option<StationConfig> {
-    let re = RegexBuilder::new(
-        r"^ATIS ([a-zA-Z- ]+) ([1-3]\d{2}(\.\d{1,3})?)(,[ ]?TRAFFIC ([1-3]\d{2}(\.\d{1,3})?))?(,[ ]?VOICE ([a-zA-Z-:]+))?(,[ ]?INFO ([a-zA-Z]))?$",
-    )
-    .case_insensitive(true)
-    .build()
-    .unwrap();
-
-    re.captures(config).map(|caps| {
-        let name = caps.get(1).unwrap().as_str();
-        let atis_freq = caps.get(2).unwrap().as_str();
-        let atis_freq = (f64::from_str(atis_freq).unwrap() * 1_000_000.0) as u64;
-        let traffic_freq = caps
-            .get(5)
-            .map(|freq| (f64::from_str(freq.as_str()).unwrap() * 1_000_000.0) as u64);
-        let tts = caps
-            .get(8)
-            .and_then(|s| TextToSpeechProvider::from_str(s.as_str()).ok());
-        let info_ltr = caps
-            .get(10)
-            .map(|ilo| ((ilo.as_str()).chars().next().unwrap().to_ascii_uppercase()) as char);
-        StationConfig {
-            name: name.to_string(),
-            atis: atis_freq,
-            traffic: traffic_freq,
-            tts,
-            info_ltr_override: info_ltr,
+    let re = RegexBuilder::new(r"^ATIS ([a-zA-Z- ]+) ([1-3]\d{2}(\.\d{1,3})?)")
+        .case_insensitive(true)
+        .build()
+        .unwrap();
+
+    let caps = re.captures(config).unwrap();
```

Comment:
Since we cannot know if this will succeed, we have to return `None` instead of panicking here

```suggestion
    let caps = re.captures(config)?;
```

(the other `unwrap`s in this method should be fine since we know that they will always succeed, but it would actually also not hurt to use `?` for them)

Reply:
Done for all the main unwraps, (along with a unit test to detect the problem in the first place).
Not having as much luck with the other optional ones, so I think I'll leave those as-is as I agree we should not get unparsable values unless the regex is changed.